File: server/app/model/inference.go
package model

import (
	"deep-ai-server/app/db"
	"time"
)

type Inference struct {
	// gorm.Job
	ID        uint `gorm:"primary_key"`
	CreatedAt time.Time
	UpdatedAt time.Time  `json:"-"`
	DeletedAt *time.Time `sql:"index" json:"-"`

	Name             string
	Description      string
	Status           int
	Framework        string
	FrameworkVersion string
	Dimension        string
	ModelFile        string

	//Num    int `json:"-"`
	//CPU    int `json:"-"`
	//Memory int `json:"-"`
	//GPU    int `json:"-"`

	ServiceName string

	UserID    uint `json:"-"`
	AIModelID uint `json:"-"`
}

const (
	INFERENCE_STATUS_PROCESSING = iota
	INFERENCE_STATUS_AVAILABLE
	INFERENCE_STATUS_UNAVAILABLE
)

const (
	INFERENCE_FRAMEWORK_TENSORFLOW = "tensorflow"
	INFERENCE_FRAMEWORK_PYTORCH    = "pytorch"
)

func (inference Inference) UpdateInferenceStatus(status int) error {
	return db.DB().Model(&inference).Update(&Inference{
		Status: status,
	}).Error
}

File: client/src/mock/inference.ts
import Mock from 'mockjs'
import baseUrl from '@/api/baseUrl'

function getInferenceList() {
    let dataList = []

    for (let i = 0; i < 30; i++) {
        dataList.push({
            'ID': Mock.Random.natural(),
            'CreatedAt': '@DATETIME("yyyy-MM-dd HH:mm:ss")',
            'UpdateAt': '@DATETIME("yyyy-MM-dd HH:mm:ss")',
            'DeleteAt': '@DATETIME("yyyy-MM-dd HH:mm:ss")',
            'Name': 'inference',
            'Description': 'inference',
            'Status|0-1': 1,
            'FrameWork': 'tf',
            'FrameWorkVersion': 'v1.0.1',
            'ModelPath': '/xxxx/xxxx',
            'Args': '-xxx',
            'Num': 0,
            'CPU': 1,
            'Memory': 100,
            'GPU': 1
        })
    }

    return dataList
}

Mock.mock(`${baseUrl.deepAI}/inference`, 'get', getInferenceList)
